SuccessEycaActivationActivity/handler.ts

```diff
@@ -0,0 +1,115 @@
+import { Context } from "@azure/functions";
+import { fromOption, toError } from "fp-ts/lib/Either";
+import { identity } from "fp-ts/lib/function";
+import { fromEither, tryCatch } from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { StatusEnum } from "../generated/definitions/CardActivatedStatus";
+import { UserEycaCardModel } from "../models/user_eyca_card";
+import { extractEycaExpirationDate } from "../utils/cgn_checks";
+import { genRandomCardCode } from "../utils/cgnCode";
+import { errorsToError } from "../utils/conversions";
+
+export const ActivityInput = t.interface({
+  fiscalCode: FiscalCode
+});
+
+export type ActivityInput = t.TypeOf<typeof ActivityInput>;
+
+// Activity result
+const ActivityResultSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+
+type ActivityResultSuccess = t.TypeOf<typeof ActivityResultSuccess>;
+
+const ActivityResultFailure = t.interface({
+  kind: t.literal("FAILURE"),
+  reason: t.string
+});
+
+type ActivityResultFailure = t.TypeOf<typeof ActivityResultFailure>;
+
+export const ActivityResult = t.taggedUnion("kind", [
+  ActivityResultSuccess,
+  ActivityResultFailure
+]);
+
+export type ActivityResult = t.TypeOf<typeof ActivityResult>;
+
+const failure = (context: Context, logPrefix: string) => (
+  err: Error,
+  description: string = ""
+) => {
+  const logMessage =
+    description === ""
+      ? `${logPrefix}|FAILURE=${err.message}`
+      : `${logPrefix}|${description}|FAILURE=${err.message}`;
+  context.log.info(logMessage);
+  return ActivityResultFailure.encode({
+    kind: "FAILURE",
+    reason: err.message
+  });
+};
```

Comment:
This function is repeated almost the same in 4 different files. Can we generalize his implementation to avoid code duplication?